perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_resource_attribute_def_def_filesLimit.java

```diff
@@ -48,30 +54,30 @@ public void checkAttributeSemantics(PerunSessionImpl perunSession, Member member
 
 		//Get FilesLimit value
 		if(attribute.getValue() != null) {
-			filesLimit = (Integer) attribute.getValue();
+			filesLimit = attribute.valueAsInteger();
 		} else {
 			try {
 				attribute = perunSession.getPerunBl().getAttributesManagerBl().getAttribute(perunSession, resource, A_R_defaultFilesLimit);
 			} catch (AttributeNotExistsException ex) {
 				throw new ConsistencyErrorException("Attribute with defaultFilesLimit from resource " + resource.getId() + " could not obtained.", ex);
 			}
 			if(attribute != null && attribute.getValue() != null) {
-				filesLimit = (Integer) attribute.getValue();
+				filesLimit = attribute.valueAsInteger();
+				if (filesLimit < 0) throw new WrongReferenceAttributeValueException(attribute, null, resource, member, attribute + " cannot be less than 0.");
```

Comment:
This condition is already in the syntax check. Thus, it is not necessary for the semantics check.

Reply:
Same here. Just ignore it... :D